### PR TITLE
Now you can approve or deny device auth on push notifications.

### PR DESCRIPTION
**Pull request by jorgeblacio**

### CODE REVIEW

File: src/main/kotlin/com/criptext/mail/androidui/CriptextNotification.kt [hunk 1]
```diff
@@ -104,6 +135,56 @@ class CriptextNotification(val ctx: Context) {
...
+                .setContentText(body)
+                .setAutoCancel(true)
+                .setGroupAlertBehavior(Notification.GROUP_ALERT_SUMMARY)
+                .setSound(defaultSound)
+                .setContentIntent(clickIntent)
+                .setGroup(ACTION_LINK_DEVICE)
+                .setGroupSummary(false)
+                .setSmallIcon(R.drawable.push_icon)
+                .addAction(R.drawable.check, "Approve", okPendingIntent)
+                .addAction(R.drawable.x, "Deny", denyPendingIntent)
```
Comment (danieltigse) [inline src/main/kotlin/com/criptext/mail/androidui/CriptextNotification.kt]: Use R.string

File: src/main/kotlin/com/criptext/mail/push/data/LinkDeviceActionService.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+
+
+    public override fun onHandleIntent(intent: Intent?) {
+        val action = intent!!.action
+        val notificationId = intent.getIntExtra("notificationId", 0)
+        when {
+            APPROVE == action -> {
+                pushDataSource?.submitRequest(PushRequest.LinkAccept(randomId, notificationId))
```
Comment (not-gabriel) [inline src/main/kotlin/com/criptext/mail/push/data/LinkDeviceActionService.kt]: `IntentService` already runs in background. There is no need for `AsyncTask`, so you could just call the Worker directly.

File: src/main/kotlin/com/criptext/mail/push/data/LinkDeviceActionService.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+
+class LinkDeviceActionService : IntentService("Link Device Action Service") {
+
+    companion object {
+        const val APPROVE = "Approve"
+        const val DENY = "Deny"
+
+        private var pushDataSource: PushDataSource? = null
+        private var randomId: String = ""
```
Comment (not-gabriel) [inline src/main/kotlin/com/criptext/mail/push/data/LinkDeviceActionService.kt]: why do you need to keep this data here? the Service should be as stateless as possible.

File: src/main/kotlin/com/criptext/mail/push/data/LinkDeviceActionService.kt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            }
+        }
+    }
+
+
+
+    public override fun onHandleIntent(intent: Intent?) {
+        val action = intent!!.action
+        val notificationId = intent.getIntExtra("notificationId", 0)
+        when {
```
Comment (not-gabriel) [inline src/main/kotlin/com/criptext/mail/push/data/LinkDeviceActionService.kt]: parsing arguments should be in it's own function.

File: src/main/kotlin/com/criptext/mail/push/data/PushAPIRequestHandler.kt [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import com.github.kittinunf.result.Result
+
+class PushAPIRequestHandler(ctx: Context, private val manager: NotificationManager){
+    private val activeAccount = ActiveAccount.loadFromStorage(ctx)!!
+    private val apiClient = PushAPIClient(HttpClient.Default(), activeAccount.jwt)
+
+    fun linkAccept(deviceId: String, notificationId: Int){
+        val operation = Result.of { apiClient.postLinkAccept(deviceId) }
+        when(operation){
+            is Result.Success -> manager.cancel(notificationId)
```
Comment (not-gabriel) [inline src/main/kotlin/com/criptext/mail/push/data/PushAPIRequestHandler.kt]: needs error handling

**Pull request merged**